internal/addrs/move_endpoint_module.go

```diff
@@ -704,3 +704,53 @@ func (r AbsResourceInstance) MoveDestination(fromMatch, toMatch *MoveEndpointInM
 		panic("unexpected object kind")
 	}
 }
+
+// IsModuleMoveReIndex takes the from and to endpoints from a move statement,
+// and returns true if the only changes are to module indexes, and all
+// non-absolute paths remain the same.
+func IsModuleMoveReIndex(from, to *MoveEndpointInModule) bool {
+	// The statements must originate from the same module.
+	if !from.module.Equal(to.module) {
+		panic("cannot compare move expressions from different modules")
+	}
+
+	switch f := from.relSubject.(type) {
+	case AbsModuleCall:
+		switch t := to.relSubject.(type) {
+		case ModuleInstance:
+			if len(t) != 1 {
+				// An AbsModuleCall only ever has one segment, so the
+				// ModuleInstance length must match.
+				return false
```

Comment:
I don't think I follow what this comment is asserting. :thinking: I don't recall any rule that an `AbsModuleCall` has a fixed number of "segments", assuming by that you mean the `ModuleInstanceStep` values inside its module path. Could you elaborate a bit more on what you meant here?

For example, it would be valid to write `from = module.foo[0].module.bar[0].module.baz` to produce an `AbsModuleCall` whose module instance is `module.foo[0].module.bar[0]` (two steps) and whose "call address" is `module.baz` (one more "semi-step"); if we also have `to = module.foo[0].module.bar[0].module.baz[0]` then I _think_ that means `len(t)` here would be 2, and thus we'd always return `false` even though my intuition says that it meeds the definition written in the function docstring above.

Reply:
ooh good catch. That was just an oversight on my part. After shuffling addresses around long enough I think I had the various separated module portions of different address types mixed up. I'll update this to compare `AbsModuleCall` addresses in their entirety.